src/main/scala/com/atomist/tree/content/text/microgrammar/MatcherMicrogrammarConstruction.scala

```diff
@@ -11,10 +11,28 @@ object MatcherMicrogrammarConstruction {
     val parsedMatcher = matcherParser.parseMatcher(name, grammar)
 
     val knownMatchers: Map[String, Matcher] = submatchers.map {
-      case (name, grammar : String) => (name, matcherParser.parseMatcher(name, grammar))
-    }
+      case (name, matcher) => (name, interpretMatcher(matcher, name))
+      }
 
     new MatcherMicrogrammar(parsedMatcher, name, knownMatchers)
   }
 
+  def constructOr(properties: Map[String, Any]): Matcher = {
+    val componentProperty = properties.getOrElse("components", throw new RuntimeException("an Or should have a component"))
+    // I have no idea why the array comes in as a map of indices to values but it does
+    val components = componentProperty match {
+      case m: Map[_, _] => m.values.map(interpretMatcher(_))
+      case _ => throw new RuntimeException("I expected an array of 'or' components to get to me as a map")
```

Comment:
I guess that's my style, which I probably picked up from Elm. If we hit this it's a bug ... or someone wrote some wrong JavaScript